google/cloud/pubsub/internal/defaults.cc

```diff
@@ -61,10 +60,14 @@ Options DefaultCommonOptions(Options opts) {
     opts.set<GrpcTracingOptionsOption>(internal::DefaultTracingOptions());
   }
   if (!opts.has<pubsub::RetryPolicyOption>()) {
-    opts.set<pubsub::RetryPolicyOption>(DefaultRetryPolicy()->clone());
+    opts.set<pubsub::RetryPolicyOption>(
+        std::make_shared<pubsub::LimitedTimeRetryPolicy>(
```

Comment:
It might be worth sticking with the `clone()` idiom rather than using `make_shared` directly.

Reply:
Done. Maybe you can explain to me why we might prefer one over the other?

Reply:
The big difference is that `clone()` is polymorphic.  While that doesn't matter in these particular cases, I reckoned that if we have `clone()` anyway then we might as well use it, even when we know the compile-time type.

Aside: I think you missed a couple of occurrences in the `test_retry_policies.cc` edits, but not worth worrying about now I'd proffer.

Reply:
yup, I missed those 🤦‍♂️